code_search/src/code_search/dataflow/transforms/github_dataset.py

```diff
@@ -44,10 +40,9 @@ def expand(self, input_or_inputs):
 
     pairs, tokenize_errors = tokenize_result.rows, tokenize_result.err
 
-    if self.target_dataset:
+    if self.failed_tokenize_table:
       (tokenize_errors  # pylint: disable=expression-not-assigned
-       | "Failed Tokenization" >> gh_bq.WriteFailedTokenizedData(self.project, self.target_dataset,
-                                                                 self.failed_tokenize_table)
+       | "Failed Tokenization" >> gh_bq.WriteFailedTokenizedData(self.failed_tokenize_table)
```

Comment:
Can we just do:

```
  beam.io.WriteToBigQuery(table=self.failed_tokenize_table,
                               schema=failed_tokenize_schema,
                               create_disposition=beam.io.BigQueryDisposition.CREATE_IF_NEEDED,
                               write_disposition=beam.io.BigQueryDisposition.WRITE_EMPTY)
```

We can replace WriteFailedTokenizedData
https://github.com/kubeflow/examples/blob/fa1311833c388b172093a82a1d8e4628c73b4fb2/code_search/src/code_search/dataflow/transforms/github_bigquery.py#L77

With an instance of [BigQuerySchema](https://github.com/kubeflow/examples/blob/fa1311833c388b172093a82a1d8e4628c73b4fb2/code_search/src/code_search/dataflow/transforms/bigquery.py#L45) 

This is a helper class for constructing a schema from a list of columns.

I think this is an improvement over the original code and I think its worth doing that rather than updating the underlying transforms.

Reply:
done